community/kernel/src/main/java/org/neo4j/kernel/impl/coreapi/PropertyContainerLocker.java

```diff
@@ -36,71 +36,70 @@
 {
     public Lock exclusiveLock( Supplier<Statement> stmtSupplier, PropertyContainer container )
```

Comment:
This method is also used by the `TopLevelTransaction` or `PlaceboTransaction` in order to implement [`Transaction.acquireWriteLock(PropertyContainer)`](https://github.com/neo4j/neo4j/blob/3.0/community/graphdb-api/src/main/java/org/neo4j/graphdb/Transaction.java#L156).

The `Supplier<Statement>` passed in through that API will actually "acquire a new" statement (the implementation of course will acquire a new statement by incrementing the reference count on the current statement, and to make things more confusing this is done via a method call called `currentStatement` despite it actually providing a _new_ statement). Thus not closing the statement in this method will cause unbalanced statements for transactions that do explicit locking.